runtime/lua/vim/shared.lua

```diff
@@ -608,7 +608,7 @@ end
 ---@see |vim.tbl_isarray()|
 ---
 ---@param t table
----@return boolean `true` if list-like table, else `false`.
+---@return boolean|nil `true` if list-like table, else `false`.
```

Comment:
Uh, so should I just silence the warning from luals?

Reply:
Maybe, but I'm not sure. Probably we may want to clean up this HACK instead; ref https://github.com/neovim/neovim/pull/16440/files#r1164167492 -- /cc @bfredl

Reply:
Surely we just change this to only return a Boolean?

Reply:
Yep, that TODO can be done now. Just delet this part of the code:
```lua
    -- TODO(bfredl): in the future, we will always be inside nvim
    -- then this check can be deleted.
    if vim._empty_dict_mt == nil then
      return nil
    end
```
and it will never return `nil`.

Reply:
OK, deleted the todo.